### PR TITLE
Add REST API events to Audit Log

### LINKED ISSUE #22352: Audit rest api update requests
### Description

Need a way to monitor updates from rest api, like persisting requests into existing log table.

### Use case/motivation

Airflow UI audit user actions by logging into log table, but this feature is missing in the rest api endpoints. For production this feature is needed to track mis usages or bugs.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by bbovenzi**
Closes https://github.com/apache/airflow/issues/22352

Add logging to all CRUD REST API endpoints. In the `action_logging` decorator, include checks for the url and origin to also distinguish between API endpoints from the Airflow UI vs externally. `api.{name_of_function}` or `ui.{name_of_function}`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/migrations/versions/0134_2_9_0_increase_log_event_string_size.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+revision = 'e54d90c77877'
+down_revision = '88344c1d9134'
+branch_labels = None
+depends_on = None
+airflow_version = '2.9.0'
+
+
+def upgrade():
+    """Apply increase Log event string size"""
+    pass
```
Comment (jedcunningham) [inline airflow/migrations/versions/0134_2_9_0_increase_log_event_string_size.py]: You need to write the migration.

File: airflow/migrations/versions/0134_2_9_0_increase_log_event_string_size.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+def downgrade():
+    """Unapply increase Log event string size"""
+    pass
```
Comment (jedcunningham) [inline airflow/migrations/versions/0134_2_9_0_increase_log_event_string_size.py]:
```suggestion
    with op.batch_alter_table("log") as batch_op:
        batch_op.alter_column("event", type_=sa.String(60))


def downgrade():
    """Unapply increase Log event string size"""
    conn = op.get_bind()
    if conn.dialect.name == "mssql":
        with op.batch_alter_table("log") as batch_op:
            batch_op.drop_index("idx_log_event")
            batch_op.alter_column("event", type_=sa.String(30), nullable=False)
            batch_op.create_index("idx_log_event", ["event"])
    else:
        with op.batch_alter_table("log") as batch_op:
            batch_op.alter_column("event", type_=sa.String(30), nullable=False)
```

try this

File: airflow/migrations/versions/0134_2_9_0_increase_log_event_string_size.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def downgrade():
+    """Unapply increase Log event string size"""
+    conn = op.get_bind()
+    if conn.dialect.name == "mssql":
+        with op.batch_alter_table("log") as batch_op:
+            batch_op.drop_index("idx_log_event")
+            batch_op.alter_column("event", type_=sa.String(30), nullable=False)
+            batch_op.create_index("idx_log_event", ["event"])
```
Comment (jscheffl) [inline airflow/migrations/versions/0134_2_9_0_increase_log_event_string_size.py]: Are you sure you want to have specifiy for MSSQLServer in here if this DB is not supported anymore?

**Review by jscheffl — APPROVED** [review]

**Comment by Lee-W** [comment]
@bbovenzi  `package.json` looks like accidentally added to me. Could you please check? Thanks!

**Review by jedcunningham — COMMENTED** [review]
Can we add a newfragment that documents the `event` changes that happen as a result of this?

File: newsfragments/37734.significant.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Add REST API actions to Audit Log events
+
+The Audit Log ``event`` name for REST API events will be prepended with ``api.`` or ``ui.``. Depending on if it came from the Airflow UI or externally
```
Comment (jedcunningham) [inline newsfragments/37734.significant.rst]:
```suggestion
The Audit Log ``event`` name for REST API events will be prepended with ``api.`` or ``ui.``, depending on if it came from the Airflow UI or externally.
```

Small nit.

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**